modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/product/navigation/user/personal/menu/SignOutUserPersonalMenuEntry.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.users.admin.web.internal.product.navigation.user.personal.menu;
+
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.Portlet;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.product.navigation.user.personal.menu.BaseUserPersonalMenuEntry;
+import com.liferay.product.navigation.user.personal.menu.UserPersonalMenuEntry;
+
+import java.util.Locale;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(
+	immediate = true,
+	property = {
+		"product.navigation.user.personal.menu.entry.order:Integer=100",
+		"product.navigation.user.personal.menu.group:Integer=400"
+	},
+	service = UserPersonalMenuEntry.class
+)
+public class SignOutUserPersonalMenuEntry extends BaseUserPersonalMenuEntry {
```

Comment:
If we remove the implementation-related methods from the interface (the ones that are not used by the caller of this API), then this class can just implement the interface directly without extending from the base class.